src/core1_task.rs

```diff
@@ -527,6 +539,41 @@ pub fn core_1_task(
             (frame_telemetry, frame_header_is_valid)
         };
 
+        //if in high power mode need to check thermal-recorder hasn't made a recording
+        if needs_ffc && !device_config.use_low_power_mode() {
+            //only check status every 20 seconds
+            if frame_telemetry.frame_num - last_rec_check > 9 * 20 {
+                if let Ok(is_recording) = shared_i2c.tc2_agent_is_recording(&mut delay) {
+                    high_power_recording = is_recording;
+                    last_rec_check = frame_telemetry.frame_num;
+                    info!(
+                        "Checking if recording {} am recording ?? {}",
+                        is_recording, high_power_recording
+                    );
+                    if high_power_recording {
+                        sio.fifo.write(Core1Task::StartRecording.into());
+                        high_power_recording = true;
+                    } else {
+                        sio.fifo.write(Core1Task::EndRecording.into());
+                        high_power_recording = false;
+                        thread_local_frame_buffer
```

Comment:
Just to confirm – we're writing the FFC imminent bytes when the recording on the pi has *ended*?  Can you briefly explain the whole mechanism at play here – just so I'm sure I understand what's going on.